src/helpers/helpers.time.js

```diff
@@ -53,25 +55,35 @@ function generateTicksNiceRange(options, dataRange, niceRange) {
 	var ticks = [];
 	if (options.maxTicks) {
 		var stepSize = options.stepSize;
-		var startTick = options.min !== undefined ? options.min : niceRange.min;
+		var startTick = helpers.isNullOrUndef(options.min) ? niceRange.min : options.min;
 		var majorUnit = options.majorUnit;
 		var majorUnitStart = majorUnit ? moment(startTick).add(1, majorUnit).startOf(majorUnit) : startTick;
 		var startRange = majorUnitStart.valueOf() - startTick;
 		var stepValue = interval[options.unit].size * stepSize;
 		var startFraction = startRange % stepValue;
 		var alignedTick = startTick;
-		if (startFraction && majorUnit && !options.timeOpts.round && !options.timeOpts.isoWeekday) {
+
+		// first tick
+		if (startFraction && majorUnit && !options.timeOpts.round && !options.timeOpts.isoWeekday && helpers.isNullOrUndef(options.min)) {
```

Comment:
My understanding of [`isoWeekDay`](http://momentjs.com/docs/#/get-set/iso-weekday/) is that's a number representing the day (1: Monday - 7: Sunday) that should be considered the first day of the week and I guess that  `isoWeekDay: true` is an alias to 1 (Monday).